src/tests/ftest/launch.py

```diff
@@ -985,21 +986,85 @@ def replace_yaml_file(yaml_file, args, yaml_dir):
             #   - found in the test yaml
             #   - have a user-specified replacement
             if key in yaml_find and user_value:
-                values_to_replace = []
                 if key.startswith("test_"):
                     # The entire server/client test yaml list entry is replaced
                     # by a new test yaml list entry, e.g.
-                    #   '- serverA' --> '- wolf-1'
-                    value_format = "- {}"
-                    values_to_replace = [value_format.format(item) for item in yaml_find[key]]
+                    #   '  test_servers: server-[1-2]' --> '  test_servers: wolf-[10-11]'
+                    #   '  test_servers: 4'            --> '  test_servers: wolf-[10-13]'
+                    if not isinstance(yaml_find[key], list):
+                        yaml_find[key] = [yaml_find[key]]
+
+                    for yaml_find_item in yaml_find[key]:
+                        replacement = NodeSet()
+                        try:
+                            # Replace integer placeholders with the number of nodes from the user
+                            # provided list equal to the quantity requested by the test yaml
+                            quantity = int(yaml_find_item)
+                            if args.override and args.test_clients:
+                                # When individual lists of server and client nodes are provided with
+                                # the override flag set use the full list of nodes specified by the
+                                # test_server/test_client arguments
+                                quantity = len(user_value)
+                            elif args.override:
+                                print(
+                                    "Warning: In order to override the node quantity a "
+                                    "'--test_clients' argument must be specified: {}: {}".format(
+                                        key, yaml_find_item))
+                            for _ in range(quantity):
+                                try:
+                                    replacement.add(user_value.pop(0))
+                                except IndexError:
+                                    # Not enough nodes provided for the replacement
+                                    if not args.override:
+                                        replacement = None
+                                    break
```

Comment:
Why not more direct?
```suggestion
                            if len(user_value) < quantity:
                                # Not enough nodes provided for the replacement
                                replacement = None
                            else:
                                replacement.add(NodeSet.from_list(user_value))
```

Reply:
Ask for forgiveness ... and its less code for the override option.